src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -176,7 +176,7 @@ private void ReadConstructorArguments(ref ReadStack state, ref Utf8JsonReader re
 
                     if (!(jsonParameterInfo!.ShouldDeserialize))
                     {
-                        reader.TrySkip();
+                        reader.Skip();
```

Comment:
Before this we were silently failing (i.e. not throwing) when there was invalid JSON as the value of a parameter that wasn't a candidate for deserialization, right? Did you add a test for this?

Reply:
No -  this code path applies only when `reader.IsFinalBlock` is true, so `reader.TrySkip` was always `true` here. This code path has full test coverage.

It would be good to have tests focused on encountering invalid JSON when deserializing objects (with both parameterless & parameterized ctors) & when deserializing when `options.ReferenceHandling.Preserve` is active, similar to work done in https://github.com/dotnet/corefx/pull/42254 & https://github.com/dotnet/corefx/pull/42066. I logged an issue to track this - https://github.com/dotnet/runtime/issues/35517.

Reply:
> This code path has full test coverage.

What's the reason for this change, then?

Reply:
Are you asking why was changed in this PR? I noticed it while skimming through the files for places where we throw when `reader.TokenType != JsonTokenType.PropertyName` (in some cases, `Debug.Assert` is fine). I thought to change it since I have context here.

---
Otherwise,

`TrySkip` is meant for when the reader might not have the data needed for the children to be skipped successfully. We know that's not the case here.

Using `Skip` provides extra validation for this assumption because `InvalidOperationException` is thrown if `IsFinalBlock` is false.

Reply:
> Using Skip provides extra validation for this assumption because InvalidOperationException is thrown if IsFinalBlock is false.

This seems reasonable to me, and its the answer to what I was trying to ask. Thanks.